includes/framework/QType.class.php

```diff
@@ -68,13 +68,23 @@ public function __construct($strMessage, $intOffset = 2) {
 			throw new QCallerException('Type should never be instantiated.  All methods and variables are publically statically accessible.');
 		}
 
+		/** String Type */
 		const String = 'string';
+		/** Integer Type */
 		const Integer = 'integer';
+		/** Float Type */
 		const Float = 'double';
+		/** Boolean Type */
 		const Boolean = 'boolean';
+		/** Object Type */
 		const Object = 'object';
+		/** Array Type */
 		const ArrayType = 'array';
+		/** JSON - String Type but valid JSON Text */
+		const Json = 'json';
```

Comment:
If we are going to support database-specific types, then we need to declare them so in the DatabaseBase class. Do you have any suggestion on how we can achieve that? 

If you think we should drop it for now, I'm okay with that so far. However, the modifications in QPostgreSqlDatabase class are required to not let Code Generation fail. We can change that behavior to 'text' treatment for JSON types. But that would be a little troublesome later in code generation. How about we get a `QDbDataType` for things like that?